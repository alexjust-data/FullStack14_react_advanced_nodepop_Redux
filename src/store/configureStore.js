import { createStore } from 'redux';
import { persistStore, persistReducer } from 'redux-persist';
import storage from 'redux-persist/lib/storage'; 
import rootReducer from './reducers'; 

const persistConfig = {
  key: 'root',
  storage,
  whitelist: ['session'] 
};

const persistedReducer = persistReducer(persistConfig, rootReducer);

export const store = createStore(persistedReducer);
export const persistor = persistStore(store);
