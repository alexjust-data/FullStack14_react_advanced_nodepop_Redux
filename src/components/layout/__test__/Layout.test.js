import React from 'react';
import { render } from '@testing-library/react';
import { MemoryRouter } from 'react-router-dom';
import { Provider } from 'react-redux';
import configureStore from 'redux-mock-store';
import Layout from '../Layout';

const mockStore = configureStore();
const store = mockStore({
  session: {
    token: null 
  },
  // Otros estados iniciales si fueran necesarios
});

describe('Layout', () => {
  it('should match snapshot', () => {
    const { asFragment } = render(
      <Provider store={store}>
        <MemoryRouter>
          <Layout />
        </MemoryRouter>
      </Provider>
    );
    expect(asFragment()).toMatchSnapshot();
  });
});

