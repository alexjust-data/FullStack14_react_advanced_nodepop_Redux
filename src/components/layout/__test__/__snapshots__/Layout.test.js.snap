// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Layout should match snapshot 1`] = `
<DocumentFragment>
  <header>
    Nodepop React
    <nav>
      <ul>
        <li>
          <a
            class=""
            href="/adverts"
          >
            Nodepop
          </a>
        </li>
        <li>
          <a
            class=""
            href="/adverts/new"
          >
            New advert
          </a>
        </li>
      </ul>
    </nav>
    <a
      href="/login"
    >
      Login
    </a>
  </header>
  <main />
  <footer>
    @Keepcoding 2021
  </footer>
</DocumentFragment>
`;
